scylla/src/transport/session.rs

```diff
@@ -650,15 +650,34 @@ impl Session {
                 |connection: Arc<Connection>,
                  consistency: Consistency,
                  execution_profile: &ExecutionProfileInner| {
+                    let default_consistency = execution_profile.serial_consistency;
                     let serial_consistency = query
                         .config
                         .serial_consistency
-                        .unwrap_or(execution_profile.serial_consistency);
+                        .unwrap_or(default_consistency);
                     // Needed to avoid moving query and values into async move block
                     let query_ref = &query;
                     let values_ref = &serialized_values;
                     let paging_state_ref = &paging_state;
+                    let values_size = values.serialized().unwrap().into_owned().size();
```

Comment:
`.serialized()` is an operation that can be potentially costly. It is already computed on line 624, so I suggest using the already computed `serialized_values` instead.